blog/2022-05-standalone-components/README.md

```diff
@@ -0,0 +1,333 @@
+Dadurch werden die Providers und Root-Routen bereitgestellt.
+Wir empfehlen, die Routendefinitionen weiterhin in einer separaten Datei unterhalb des Ordners `src/app` aufzubewahren:
+
+```ts
+// app.routes.ts
+export const appRoutes: Routes = [
+  { path: 'books', component: DashboardComponent },
+  { path: 'books/:isbn', component: BookDetailsComponent },
+];
+```
+
+```ts
+// main.ts
+// ...
+import { appRoutes } from './app/app.routes';
+
+bootstrapApplication(AppComponent, {
+  providers: [
+    importProvidersFrom(RouterModule.forRoot(appRoutes)),
+  ]
+}).catch(err => console.error(err));
+```
+
+Um diesen Schritt zu vereinfachen, wird Angular vermutlich in Zukunft selbst eine Funktion wie `provideRouter()` anbieten, die nur die Providers ohne ein Modul liefert.
+Auch für das Framework NgRx sind Funktionen wie `provideStore()` oder `provideEffects()` denkbar.
+
+
+## Direktiven des Routers nutzen
+
+Wenn wir im Template einer Standalone-Komponente die Direktiven des Routers nutzen wollen, z. B. `RouterLink` oder `RouterOutlet`, müssen wir das `RouterModule` dort importieren.
+
+```ts
+@Component({
+  // ...
+  standalone: true,
+  imports: [CommonModule, RouterModule]
+})
+export class AppComponent {}
+```
+
+
+## Lazy Loading
+
+Beim Lazy Loading mit dem Router werden für eine definierte Basisroute die Kindrouten aus einem anderen Modul nachgeladen.
+Dieses zu ladende Kindmodul wird in ein eigenes Bundle verpackt, das erst zur Laufzeit asynchron nachgeladen wird.
+Mit Modulen kann die Basisroute für Lazy Loading wie folgt definiert werden. `loadChildren` verweist auf ein Feature-Modul:
+
+```ts
+// mit NgModule:
+// loadChildren lädt ein Modul
+{
+  path: 'books',
+  loadChildren: () => import('./books/books.module').then(m => m.BooksModule)
+}
+```
+
+
+Mit Standalone Components funktioniert dieses Konzept sehr ähnlich – nur ohne Module.
+Wir definieren ebenfalls in den App-Routen eine Basisroute.
+Die Eigenschaft `loadChildren` verweist nun allerdings nur auf ein Array von Routen:
+
+
+```ts
+// books/books.routes.ts
+export const booksRoutes: Routes = [
+  { path: '', component: DashboardComponent },
+  { path: ':isbn', component: BookDetailsComponent },
+];
+```
+
+```ts
+// app.routes.ts
+// ...
+
+// mit Standalone Components:
+// loadChildren lädt ein Array von Routen
+{
+  path: 'books',
+  loadChildren: () => import('./books/books.routes').then(m => m.booksRoutes)
+}
+```
+
+Neu ist außerdem die Möglichkeit, eine einzelne Komponente mittels Lazy Loading direkt zu laden.
+Das ist besonders praktisch, wenn es sich bei der gerouteten Komponente gar nicht um ein komplexes Feature mit mehreren Kind-Routen handelt, sondern nur um eine einzelne Ansicht.
+Um eine Komponente zu laden, nutzen wir `loadComponent`:
+
+```ts
+{
+  path: 'dashboard',
+  loadComponent: () => import('./dashboard/dashboard.component').then(m => m.DashboardComponent)
```

Comment:
```suggestion
  loadComponent: () => import('./dashboard/dashboard.component').then(c => c.DashboardComponent)
```

Ich würde es `c` für Component an der Stelle nennen.

Reply:
Es ist ja ein ES-Module, das man da importiert. In der Angular-Doku nennen sie die Variable sogar `mod`: https://next.angular.io/guide/standalone-components#lazy-loading-a-standalone-component

Reply:
Ah okay ja stimmt. Dann macht das natürlich weiterhin Sinn.